data/transaction.go

```diff
@@ -122,6 +123,27 @@ type ResponseTransaction struct {
 	Code  string                  `json:"code"`
 }
 
+// TransactionSimulationResponseData holds the results of a transaction's simulation
+type TransactionSimulationResults struct {
+	Status     core.TransactionStatus                         `json:"status,omitempty"`
+	FailReason string                                         `json:"failReason,omitempty"`
+	ScResults  map[string]*transaction.SmartContractResultApi `json:"scResults,omitempty"`
```

Comment:
We might have small issues in the future with serializing this `map[string]`, but ok, can stay like this for the moment.

Reply:
you mean that a slice would have been safer? is this related to the order?